File: app/src/main/java/com/geridea/trentastico/gui/views/requestloader/TerminalMessage.java
package com.geridea.trentastico.gui.views.requestloader;


/*
 * Created with ♥ by Slava on 28/03/2017.
 */

public class TerminalMessage implements ILoadingMessage {

    private final int id;

    public TerminalMessage(int id) {
        this.id = id;
    }

    @Override
    public void process(RequestLoaderView requestLoaderView) {
        requestLoaderView.removeMessage(getMessageId());
    }

    public int getMessageId() {
        return id;
    }
}

File: app/src/main/java/com/geridea/trentastico/gui/views/requestloader/TerminalMessage.kt
package com.geridea.trentastico.gui.views.requestloader


/*
 * Created with ♥ by Slava on 28/03/2017.
 */

open class TerminalMessage(private val messageId: Int) : ILoadingMessage {

    override fun process(requestLoaderView: RequestLoaderView) = requestLoaderView.removeMessage(messageId)
}
